### PR TITLE
Minor fixes for DSL SQL

### PR DESCRIPTION
**Pull request by iemejia**
- Make the example execution runner agnostic and some minor fixes.
- Enable findbugs for the profile release and fix current issues.

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by iemejia** [comment]
R: @XuMingmin or @takidau

**Comment by iemejia** [comment]
Actually it is, all these fixes come from me trying to run the example.
The current dependency is runtime but it does not resolve the runner appropriately (notice also the missing parent call), notice alse that the profile is made default but I also put it as a first step in case someone else want to test the SQL with a different runner.

**Comment by jbonofre** [comment]
CC: @jbonofre

File: dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/arithmetic/BeamSqlArithmeticExpression.java [hunk 1]
```diff
@@ -80,12 +80,10 @@ public BeamSqlArithmeticExpression(List<BeamSqlExpression> operands) {
...
   private double getDouble(BeamSqlRow inputRecord, BeamSqlExpression op) {
     Object raw = op.evaluate(inputRecord).getValue();
-    Double ret = null;
     if (SqlTypeName.NUMERIC_TYPES.contains(op.getOutputType())) {
-      ret = ((Number) raw).doubleValue();
+      return ((Number) raw).doubleValue();
     }
-
-    return ret;
+    return 0d;
```
Comment (takidau) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/arithmetic/BeamSqlArithmeticExpression.java]: Is returning 0 really the right thing to do here? NullPointerException probably isn't ideal, either. How about throwing an IllegalStateException? Or are there cases where we actually expect this to case to happen and also be valid?

Comment (iemejia) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/arithmetic/BeamSqlArithmeticExpression.java]: For the error with Hamcrest I have the same issue and it was only solved until I did a full `mvn clean install` from the beam parent directory, then the execution worked flawlessly (capricious Maven).
For a quick execution I did:
```
mvn clean install  -DskipTests -Dcheckstyle.skip=true -Denforcer.skip=true -Dmaven.javadoc.skip=true -pl '!sdks/python' -pl '!sdks/java/maven-archetypes/examples-java8' -pl '!sdks/java/maven-archetypes/starter' -pl '!sdks/java/maven-archetypes/examples'
```

Comment (iemejia) [inline dsls/sql/src/main/java/org/apache/beam/dsls/sql/interpreter/operator/arithmetic/BeamSqlArithmeticExpression.java]: You have a point with the return, I also hesitated, probably IllegalStateException makes more sense, I am going to change, also I fixed the case of division that didn't validate division by zero, I am going to add this here too. Thanks for the review.

**Comment by takidau** [comment]
The install did the the trick, thanks! LGTM. Merging.

**Comment by takidau** [comment]
Merged. Feel free to close.

**Comment by iemejia** [comment]
Thanks @takidau.

**Pull request closed without merging**